Feature/4

dipteewarudkar opened the pull request:
@duergner   For your review 

Ref : Issue #4
1)Added Address Response/Request objects for create/get/update
2)Address AddressService to fetch/create/update address in zalando
3)Added Abstract Service same as already existing (Might need some changes here)

PS : Will not merge it with master.

dipteewarudkar committed:
Adding model classes for Address
Adding service class for Address

TODO common Abstract Class

dipteewarudkar committed:
Fixing GetAddressesResponse structure

dipteewarudkar committed:
Fixing build warnings

dipteewarudkar committed:
Fixing update address to http PUT request

dipteewarudkar committed:
Adding wrapper for zalando Api for
Cart
checkout
order
customer

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Address.java:
```diff
@@ -0,0 +1,107 @@
+public abstract class Address extends Base {
+
+	@JsonProperty
+	private String id;	
```
Please use empty lines between the single properties

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Address.java:
```diff
@@ -0,0 +1,107 @@
+	@JsonProperty
+	private String id;	
+	@JsonProperty
+	private String customer_number;
```
Please use camel case notation for property names; use the `value` property on the @JsonProperty annotation to specify a different JSON name

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/AddressCheckRequest.java:
```diff
@@ -0,0 +1,8 @@
+package io.sprucehill.zalando.api.nativecart.model;
+
+public class AddressCheckRequest extends Address{
```
What's the use for this class as it's basically just an Address?

duergner replied:
Use a space before the `{`

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/AddressCheckRequest.java:
```diff
@@ -0,0 +1,8 @@
+
+public class AddressCheckRequest extends Address{
+
+	public AddressCheckRequest(String gender,String first_name,String last_name,String street,String additional,String zip,String city,String country_code,Boolean pack_station,Boolean default_billing,Boolean default_shipping){
```
Use camel case for the parameters

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Address.java:
```diff
@@ -0,0 +1,107 @@
+
+	public Address(){}
+	
+	public Address(String id, String customer_number, String gender, String first_name, String last_name, String street,String additional, String zip, String city, String country_code, Boolean pack_station,Boolean default_billing, Boolean default_shipping) {
```
Use camel case for the parameters

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/AddressCheckResponse.java:
```diff
@@ -0,0 +1,38 @@
+ *
+ */
+public class AddressCheckResponse extends Base {
+	private String status;
```
Use @JsonProperty annotation for fields that are mapped from JSON

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/AddressCheckResponse.java:
```diff
@@ -0,0 +1,38 @@
+public class AddressCheckResponse extends Base {
+	private String status;
+	private Boolean blacklisted;
+	Address address;
```
Use private (or protected) visibility

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/AddressCheckResponse.java:
```diff
@@ -0,0 +1,38 @@
+	private String status;
+	private Boolean blacklisted;
+	Address address;
+	Address normalized_address;
```
Use camel case notation

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Base.java:
```diff
@@ -0,0 +1,12 @@
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(value = JsonInclude.Include.NON_NULL)
+public abstract class Base {
```
Simply use the existing `Base` class instead of copying it over

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Cart.java:
```diff
@@ -0,0 +1,68 @@
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public abstract class Cart extends Base{
```
Missing JavaDoc header

duergner replied:
Missing whitespace before the `{`

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Cart.java:
```diff
@@ -0,0 +1,68 @@
+
+public abstract class Cart extends Base{
+	@JsonProperty
+	private String id ;
```
Use empty line between properties

duergner replied:
Remove the whitespace before the `;`

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Cart.java:
```diff
@@ -0,0 +1,68 @@
+	public String getCartUrl() {
+		return cartUrl;
+	}
+	
```
Remove unnecessary blank lines

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Checkout.java:
```diff
@@ -0,0 +1,70 @@
+ *
+ */
+public abstract class Checkout extends Base{
+	@JsonProperty
```
Use blank lines between properties

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Checkout.java:
```diff
@@ -0,0 +1,70 @@
+	@JsonProperty
+	private String id;	
+	@JsonProperty
+	private String customer_number;
```
Use camel case notation

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Checkout.java:
```diff
@@ -0,0 +1,70 @@
+
+	public Checkout(){}
+
+	public Checkout(String id,String customer_number,String cart_id, String billing_address_id,String shipping_address_id,Delivery delivery,Payment payment){
```
Use camel case notation

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/CreateAddressRequest.java:
```diff
@@ -0,0 +1,13 @@
+ * @author dipteewarudkar
+ *
+ */
+public class CreateAddressRequest extends Address{
```
What's the use for that class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/CreateAddressResponse.java:
```diff
@@ -0,0 +1,9 @@
+ * @author dipteewarudkar
+ *
+ */
+public class CreateAddressResponse extends Address{
```
What's the use for that class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/CreateCartRequest.java:
```diff
@@ -0,0 +1,15 @@
+ * @author dipteewarudkar
+ *
+ */
+public class CreateCartRequest extends Cart{
```
What's the use for that class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/CreateCartResponse.java:
```diff
@@ -0,0 +1,11 @@
+ * @author dipteewarudkar
+ *
+ */
+public class CreateCartResponse extends Cart {
```
What's the use for that class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/CreateCheckoutRequest.java:
```diff
@@ -0,0 +1,14 @@
+ * @author dipteewarudkar
+ *
+ */
+public class CreateCheckoutRequest extends Checkout {
```
What's the use for that class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/CreateCheckoutResponse.java:
```diff
@@ -0,0 +1,5 @@
+package io.sprucehill.zalando.api.nativecart.model;
+
+public class CreateCheckoutResponse extends Checkout{
```
What's use for that class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/CreateOrderRequest.java:
```diff
@@ -0,0 +1,18 @@
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class CreateOrderRequest extends Base{
```
Missing JavaDoc header

duergner replied:
Missing whitespace before `{`

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/CreateOrderRequest.java:
```diff
@@ -0,0 +1,18 @@
+public class CreateOrderRequest extends Base{
+	
+	@JsonProperty
+	private String checkout_id;
```
Use camel case

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/CreateOrderRequest.java:
```diff
@@ -0,0 +1,18 @@
+		return checkout_id;
+	}
+
+	public void setCheckout_id(String checkout_id) {
```
Use camel case

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/CreateOrderResponse.java:
```diff
@@ -0,0 +1,10 @@
+ * @author dipteewarudkar
+ *
+ */
+public class CreateOrderResponse extends Order{
```
What's the use for that class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Delivery.java:
```diff
@@ -0,0 +1,28 @@
+ * @author dipteewarudkar
+ *
+ */
+public class Delivery extends Base{
```
Missing whitespace before `{`

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Delivery.java:
```diff
@@ -0,0 +1,28 @@
+ */
+public class Delivery extends Base{
+	@JsonProperty
+	private String earliest;
```
Missing blank line between properties

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/GetAddressResponse.java:
```diff
@@ -0,0 +1,9 @@
+ * @author dipteewarudkar
+ *
+ */
+public class GetAddressResponse extends Address {
```
What's the use for that class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/GetAddressesResponse.java:
```diff
@@ -0,0 +1,22 @@
+ */
+public class GetAddressesResponse extends Base {
+	
+	List<GetAddressResponse> addresses;
```
Missing @JsonProperty annotation

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/GetCartResponse.java:
```diff
@@ -0,0 +1,63 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class GetCartResponse  extends Base{
```
Missing JavaDoc header

duergner replied:
Missing whitespace before `{`

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/GetCartResponse.java:
```diff
@@ -0,0 +1,63 @@
+import java.util.List;
+
+public class GetCartResponse  extends Base{
+	private String id = null;
```
Missing @JsonProperty annotation

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/GetCartResponse.java:
```diff
@@ -0,0 +1,63 @@
+
+public class GetCartResponse  extends Base{
+	private String id = null;
+	private String appDomainId = null;
```
Missing blank line between the properties

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/GetCheckoutResponse.java:
```diff
@@ -0,0 +1,10 @@
+ * @author dipteewarudkar
+ *
+ */
+public class GetCheckoutResponse  extends Checkout{
```
What's the use for this class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/GetCustomerResponse.java:
```diff
@@ -0,0 +1,38 @@
+public class GetCustomerResponse extends Base {
+	
+	@JsonProperty
+	private String customer_number;
```
Missing blank line between properties

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Items.java:
```diff
@@ -0,0 +1,28 @@
+ *
+ */
+public class Items extends Base{
+	@JsonProperty
```
Missing blank line between properties

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Money.java:
```diff
@@ -0,0 +1,30 @@
+ *
+ */
+public class Money extends Base{
+	@JsonProperty
```
Missing blank line between properties

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Order.java:
```diff
@@ -0,0 +1,81 @@
+ *
+ */
+public abstract class Order extends Base{
+	@JsonProperty
```
Missing blank lines between properties

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Order.java:
```diff
@@ -0,0 +1,81 @@
+ */
+public abstract class Order extends Base{
+	@JsonProperty
+	private String order_number;	
```
Use camel case notation

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Order.java:
```diff
@@ -0,0 +1,81 @@
+	
+	public Order(){}
+	
+	public Order(String order_number,String customer_number,Address billing_address,Address shipping_address,Delivery delivery,
```
Use camel case notation

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Payment.java:
```diff
@@ -0,0 +1,50 @@
+ *
+ */
+public class Payment extends Base{
+	@JsonProperty
```
Missing blank lines between properties

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Payment.java:
```diff
@@ -0,0 +1,50 @@
+ * @author dipteewarudkar
+ *
+ */
+public class Payment extends Base{
```
Missing whitespace before `{`

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/UpdateAddressRequest.java:
```diff
@@ -0,0 +1,12 @@
+ * @author dipteewarudkar
+ *
+ */
+public class UpdateAddressRequest extends Address {
```
What's the use for this class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/UpdateAddressResponse.java:
```diff
@@ -0,0 +1,10 @@
+ * @author dipteewarudkar
+ *
+ */
+public class UpdateAddressResponse extends Address{
```
What's the use for this class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/UpdateCartItemsRequest.java:
```diff
@@ -0,0 +1,15 @@
+ * @author dipteewarudkar
+ *
+ */
+public class UpdateCartItemsRequest extends Cart{
```
What's the use for this class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/UpdateCartItemsResponse.java:
```diff
@@ -0,0 +1,10 @@
+ * @author dipteewarudkar
+ *
+ */
+public class UpdateCartItemsResponse extends Cart {
```
What's the use for this class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/UpdateCheckoutRequest.java:
```diff
@@ -0,0 +1,14 @@
+ * @author dipteewarudkar
+ *
+ */
+public class UpdateCheckoutRequest extends Checkout{
```
What's the use for this class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/UpdateCheckoutRespone.java:
```diff
@@ -0,0 +1,5 @@
+package io.sprucehill.zalando.api.nativecart.model;
+
+public class UpdateCheckoutRespone extends Checkout {
```
What's the use for this class?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AbstractService.java:
```diff
@@ -0,0 +1,285 @@
+/**
+ * @author Michael Duergner
+ */
+public abstract class AbstractService {
```
Why did you copy this class?

dipteewarudkar replied:
I am not really imagine the changes as of now that is required in this class, so copied it for now , when we have a testable version of APIs i will merge/work on this class.

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AddressService.java:
```diff
@@ -0,0 +1,77 @@
+
+	@PostConstruct
+	@Override
+	public void postConstruct() {
```
Why did you override the method but not add custom logic?

dipteewarudkar replied:
I am not really imagine the changes as of now in the AbstractService, i would remove it for now ,when we have a testable version of APIs i will merge/work on this case.

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AddressService.java:
```diff
@@ -0,0 +1,77 @@
+	}
+
+	@Override
+	public GetAddressesResponse read(String customer_number) {
```
Use camel case

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AddressService.java:
```diff
@@ -0,0 +1,77 @@
+		try {
+			request.setEntity(new StringEntity(objectMapper.writeValueAsString(updateAddressRequest)));
+		} catch (UnsupportedEncodingException e) {
+			e.printStackTrace();
```
Use logger instead of `e.printStackTrace()`

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/CartService.java:
```diff
@@ -0,0 +1,62 @@
+ * @author dipteewarudkar
+ *
+ */
+public class CartService extends AbstractService implements ICartService{
```
Missing whitespace before `{`

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/IAddressService.java:
```diff
@@ -0,0 +1,56 @@
+	 * @param customer_number The customer unique id
+	 * @return The list of user addresses
+	 */
+	GetAddressesResponse read(String customer_number) ;
```
Use camel case

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/ICartService.java:
```diff
@@ -0,0 +1,40 @@
+	 * @param createCartRequest The Cart object to be created
+	 * @return Newly created cart object
+	 */
+	CreateCartResponse create(String access_token,CreateCartRequest createCartRequest);
```
Use camel case

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/ICustomerService.java:
```diff
@@ -0,0 +1,18 @@
+	 * @param customer_number  The customer unique id
+	 * @return Details of the customer with specified customer_number
+	 */
+	GetCustomerResponse  read(String customer_number);
```
Use camel case

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/IOrderService.java:
```diff
@@ -0,0 +1,21 @@
+	 * @param createOrderRequest The order object to be created
+	 * @return The newly create order object
+	 */
+	CreateOrderResponse create(String customer_number,CreateOrderRequest createOrderRequest);
```
Use camel case

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/ICheckoutService.java:
```diff
@@ -0,0 +1,41 @@
+	 * @param checkout_id The ID of the checkout to read
+	 * @return The checkout object with the requested ID for the specified customer
+	 */
+	GetCheckoutResponse read(String customer_number,String checkout_id);
```
Use camel case

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/OrderService.java:
```diff
@@ -0,0 +1,31 @@
+public class OrderService extends AbstractService implements IOrderService{
+
+	@Override
+	public CreateOrderResponse create(String customer_number, CreateOrderRequest createOrderRequest) {
```
Use camel case

dipteewarudkar committed:
Fixing :
Formatting problems
Naming issues
Removing repeated objects

TODO AbstractService/postContructs -> Will keep this in hold till we
have a running version to test these APIs.

dipteewarudkar committed:
Formatting fixes

dipteewarudkar committed:
Changing response of read(customer_no) in addressService to
List<Address>

dipteewarudkar committed:
Fixing build javadoc errors

dipteewarudkar committed:
Fixing some more javadoc build error

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Customer.java:
```diff
@@ -0,0 +1,47 @@
+public class Customer extends Base {
+	
+	@JsonProperty
+	private String customer_number;
```
Use camel case notation

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Money.java:
```diff
@@ -0,0 +1,35 @@
+	private BigInteger amount;
+	
+	@JsonProperty
+	private String currency;
```
I think we should best use an enum here?

duergner commented:
And please create a pull request against `develop` instead of `master` for really merging it in than (I'll take care of the merging than)

dipteewarudkar committed:
1)Changing Currency to enum
2)Fixing naming bug.

dipteewarudkar committed:
Adding service for getting access token and some other small changes

dipteewarudkar committed:
Some changes to removing CreateOrderRequest and using the common Order
class.
Adding token property on Payment
Some Camel case fixes

dipteewarudkar committed:
Using Abstract Service from internal package and handling Exceptions

dipteewarudkar committed:
Some changes needed while integrating with live zalando Apis

dipteewarudkar committed:
Changing StringEntity to ByteEntity to save special characters such as 
(ü) when posting data

Adding "problem" class to get details of the exceptions

Changing the api path for adding updating cart

Passing accessToken to get customer details

Adding clientId and clientCredentials in abstractservice

dipteewarudkar committed:
Fixing javadoc build errors

dipteewarudkar committed:
Fixing some more javadoc build errors

dipteewarudkar committed:
Changing StringEntity to ByteArrayEntity

dipteewarudkar commented:
@duergner  For your review. Changes done while integration with staging zalando APis

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AccessTokenService.java:
```diff
@@ -0,0 +1,37 @@
+
+	@Override
+	public TokenInfo read() throws Exception {
+		HttpPost request = postRequest("/access_token?realm=%2Fcustomers&grant_type=client_credentials&client_id="+clientId+"&client_secret="+clientCredential);
```
Please use `NameValuePair` to create the correct body / request parameters instead of doing String concatenation

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AccessTokenService.java:
```diff
@@ -0,0 +1,37 @@
+	@Override
+	public TokenInfo read() throws Exception {
+		HttpPost request = postRequest("/access_token?realm=%2Fcustomers&grant_type=client_credentials&client_id="+clientId+"&client_secret="+clientCredential);
+		request.setHeader("Content-Type","application/x-www-form-urlencoded");
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AccessTokenService.java:
```diff
@@ -0,0 +1,37 @@
+	
+	@Override
+	public TokenInfo readFromAuthorizationCode(String authCode,String redirectUrl) throws Exception {
+		HttpPost request = postRequest("/access_token?realm=%2Fcustomers&grant_type=authorization_code&client_id="+clientId+"&client_secret="+clientCredential+"&code="+authCode+"&redirect_uri="+redirectUrl);
```
Please use `NameValuePair` to create the correct body / request parameters instead of doing String concatenation

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AccessTokenService.java:
```diff
@@ -0,0 +1,37 @@
+	@Override
+	public TokenInfo readFromAuthorizationCode(String authCode,String redirectUrl) throws Exception {
+		HttpPost request = postRequest("/access_token?realm=%2Fcustomers&grant_type=authorization_code&client_id="+clientId+"&client_secret="+clientCredential+"&code="+authCode+"&redirect_uri="+redirectUrl);
+		request.setHeader("Content-Type","application/x-www-form-urlencoded");
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AddressService.java:
```diff
@@ -0,0 +1,73 @@
+	@Override
+	public List<Address> read(String accessToken,String customerNumber) throws Exception {
+		HttpGet request = getRequest("/customers/" + customerNumber +"/addresses");
+		request.setHeader("Authorization","Bearer "+accessToken);
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AddressService.java:
```diff
@@ -0,0 +1,73 @@
+	@Override
+	public Address read(String accessToken,String customerNumber, String addressId) throws Exception {
+		HttpGet request = getRequest("/customers/" + customerNumber +"/addresses/"+addressId);
+		request.setHeader("Authorization","Bearer "+accessToken);
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AddressService.java:
```diff
@@ -0,0 +1,73 @@
+		HttpPut request = putRequest("/customers/" + customerNumber +"/addresses/"+addressId);
+		
+		try {
+			request.setHeader("Authorization","Bearer "+accessToken);
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AddressService.java:
```diff
@@ -0,0 +1,73 @@
+		
+		try {
+			request.setHeader("Authorization","Bearer "+accessToken);
+			request.addHeader("Content-Type","application/x.zalando.customer.address.update+json");
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AddressService.java:
```diff
@@ -0,0 +1,73 @@
+	@Override
+	public Address create(String accessToken,String customerNumber, Address createAddressRequest) throws Exception {
+		HttpPost request = postRequest("/customers/" + customerNumber+"/addresses");
+		request.setHeader("Authorization","Bearer "+accessToken);
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AddressService.java:
```diff
@@ -0,0 +1,73 @@
+	public Address create(String accessToken,String customerNumber, Address createAddressRequest) throws Exception {
+		HttpPost request = postRequest("/customers/" + customerNumber+"/addresses");
+		request.setHeader("Authorization","Bearer "+accessToken);
+		request.addHeader("Content-Type","application/x.zalando.customer.address.create+json");
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AddressService.java:
```diff
@@ -0,0 +1,73 @@
+	public AddressCheckResponse checkAddress(String accessToken,AddressCheckRequest checkAddressRequest) throws Exception {
+		HttpPost request = postRequest("/address-checks");
+		try {
+			request.setHeader("Authorization","Bearer "+accessToken);
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AddressService.java:
```diff
@@ -0,0 +1,73 @@
+		HttpPost request = postRequest("/address-checks");
+		try {
+			request.setHeader("Authorization","Bearer "+accessToken);
+			request.addHeader("Content-Type","application/x.zalando.address-check.create+json");
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/CartService.java:
```diff
@@ -0,0 +1,58 @@
+		try {
+			
+			request.addHeader(HttpHeaders.AUTHORIZATION,"Bearer "+accessToken);
+			request.addHeader("Content-Type","application/x.zalando.cart.create+json");
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/CartService.java:
```diff
@@ -0,0 +1,58 @@
+	@Override
+	public Cart read(String access_token, String cart_id) throws Exception {
+		HttpGet request = getRequest("/carts/" + cart_id);
+		request.addHeader("Content-Type","application/x.zalando.cart+json");
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/CartService.java:
```diff
@@ -0,0 +1,58 @@
+
+		try {
+			request.addHeader(HttpHeaders.AUTHORIZATION,"Bearer "+access_token);
+			request.addHeader("Content-Type","application/x.zalando.cart.items.update+json");
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/CheckoutService.java:
```diff
@@ -0,0 +1,50 @@
+	@Override
+	public Checkout read(String customerNumber, String checkout_id) throws Exception {
+		HttpGet request = getRequest("/customer/"+customerNumber+"/checkouts/"+checkout_id);
+		return execute(request, new TypeReference<Checkout>() {});
```
Don't we need authorisation here?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/CheckoutService.java:
```diff
@@ -0,0 +1,50 @@
+	@Override
+	public Checkout create(String customerNumber, Checkout createcheckoutRequest) throws Exception {
+		HttpPost request = postRequest("/customer/"+customerNumber+"/checkouts");
+
```
Don't we need authorisation here?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/CheckoutService.java:
```diff
@@ -0,0 +1,50 @@
+	@Override
+	public Checkout update(String customerNumber, String checkoutId,Checkout updateCheckoutRequest) throws Exception {
+		HttpPut request = putRequest("/customer/"+customerNumber+"/checkouts/"+checkoutId);
+
```
Don't we need authorisation here?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/CustomerService.java:
```diff
@@ -0,0 +1,21 @@
+	@Override
+	public Customer read(String accessToken ,String customerNumber)  throws Exception {
+		HttpGet request = getRequest("/customers/" + customerNumber);
+		request.addHeader("Authorization","Bearer "+accessToken);
```
Use the constants from `HttpHeaders` instead of "magic numbers"

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/OrderService.java:
```diff
@@ -0,0 +1,27 @@
+	@Override
+	public Order create(String customerNumber, Order createOrderRequest) throws Exception {
+		HttpPost request = postRequest("/customer/" + customerNumber +"/orders");
+		try {
```
Don't we need authorisation here?

dipteewarudkar committed:
Implemented Review  comments
1) Changed magic numbers to HttpHeaders
2)Added authorization on checkout and order services

dipteewarudkar committed:
Implementing review comments , removing content-type setting 
and adding /customers instead of encoded value

dipteewarudkar committed:
Some bug fix came up while integration.
- Changing app_domain_id to sales channel.
- Linking corrrect property to shipping address id in checkout object 
- Updating the api url for checkout and order

dipteewarudkar committed:
Removing /customers/customer_number/ from urls

duergner committed:
Should bump the version to correct snapshot one
